target/linux/ath79/dts/ar9344_mikrotik_routerboard-sxt-5n.dtsi

```diff
@@ -99,9 +99,9 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
-			partition@0 {
+			routerboot: partition@0 {
 				label = "RouterBoot";
-				reg = <0x0 0x20000>;
+				reg = <0x0 0x0>;
```

Comment:
@f00b4r0 Is setting the size to 0x0 here harmless? The last partition for a 64k chip would be soft_config then which has no size as well ...

Reply:
now, this is invalid. The top-level "Routerboot" partition is a "fixed-partitions" style partition and its start offset and size must be correctly set.

Reply:
As @f00b4r0 said, fixed means no size auto-detection guaranteed ever, so it could break in the future. For reference the flash map:
````
dev:    size   erasesize  name
mtd0: 00040000 00020000 "booter"
mtd1: 003c0000 00020000 "kernel"
mtd2: 07c00000 00020000 "ubi"
mtd3: 00010000 00001000 "RouterBoot"
mtd4: 0000c000 00001000 "bootloader1"
mtd5: 00001000 00001000 "hard_config"
mtd6: 00001000 00001000 "bios"
mtd7: 00001000 00001000 "soft_config"
````

Reply:
Actual size for each device is set now.